### PR TITLE
HPCC-21279 Audit user access to WsDfuEx.DFUFileAccess

### PR DESCRIPTION
**Pull request by wangkx**
Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
@jakesmith please review.

File: esp/services/ws_dfu/ws_dfuService.cpp [hunk 1]
```diff
@@ -6123,6 +6123,12 @@ void CWsDfuEx::getFileAccess(IEspContext &context, IUserDescriptor *udesc, SecAc
     accessInfo.setExpiryTime(expiryTime);
     accessInfo.setFileAccessPort(port);
     accessInfo.setFileAccessSSL(secure);
+
+    StringBuffer userName;
+    if (udesc)
+        udesc->getUserName(userName);
+    LOG(daliAuditLogCat,",FileAccess,,READ,,%s,%s,%s,%s,%d", userName.str(), fileName.str(),
+        req.getCluster(), req.getJobId(), req.getExpirySeconds());
```
Comment (jakesmith) [inline esp/services/ws_dfu/ws_dfuService.cpp]: Example from an existing audit log for a READ:
```
,FileAccess,HThor,READ,hthor,regress,regress::single::c.dg_fetchfilepreloadindexed,W20190125-160407,graph1
```
HThor is the component type. So should be "EspProcess" here.
'hthor' is the cluster name. Should be req.getCluster() 

The fields after the filename are normally graph name, timing, etc.
You don't have those and can't add them, but we need some way to tell what fields being added are going forward.
I suggest adding a field name as a prefix, e.g. ",jobid=blah,expirySecs=10"

**Comment by jakesmith** [comment]
@wangkx - 1 comment.

Is there a separate issue to add similar for publish?

**Comment by wangkx** [comment]
There is a LOG(daliAuditLogCat...) for publish.

**Comment by jakesmith** [comment]
@wangkx - looks good.

**Comment by jakesmith** [comment]
@richardkchapman - ready to merge.

**Pull request merged**